src/PrestaShopBundle/Twig/AdminExtension.php

```diff
@@ -66,23 +79,44 @@ public function onKernelRequest(GetResponseEvent $event)
             // Construct tabs and inject into twig tpl
             $tabDataContent = [];
             // Get current route name to know when to put "current" class on HTML dom
-            $currentRouteName = $this->request->get('_route');
+            $currentRouteName = $parameterBag->get('_route');
 
             foreach ($tabConfiguration[$controllerName] as $tabName => $tabData) {
                 $tabData['isCurrent'] = false;
                 if ($currentRouteName === $tabData['route']) {
                     $tabData['isCurrent'] = true;
                 }
-                $tabData['title'] = $this->translator->trans($tabData['title'], [], 'AdminControllersListener');
-                $tabData['route'] = $this->router->generate($tabData['route']);
-                $tabDataContent[] = $this->renderer->render(
+
+                $tabDataContent[] = $this->environment->render(
                     'PrestaShopBundle:Admin/Common/_partials:_header_tab.html.twig',
                     ['tabData' => $tabData]
                 );
             }
             // Inject them to templating system as global to be able to pass it to the legacy afterwards and once
             // controller has given a response
-            $this->renderer->addGlobal('headerTabContent', $tabDataContent);
+            return $tabDataContent;
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getGlobals()
+    {
+        $globals = array();
+
+        if (null !== $this->requestStack->getCurrentRequest()) {
```

Comment:
hmm you can do even better: only build and return it in "admin" back office.
Sounds like a premature optimization but I'm pretty sure no one will come back here ^_^

But sounds like boring as you need to load ``admin_routing.yml`` file and then try to match the current route with routerMatcher: I'm not sure it's worth from a performance point of view: what do you think ?